From 3768fdb7585cb6d8f8330964a3a21bee27ed3f6e Mon Sep 17 00:00:00 2001
From: "Demakkanavar, Kenchappa" <kenchappa.demakkanavar@intel.com>
Date: Fri, 26 Feb 2021 16:07:52 +0530
Subject: [PATCH 056/109] dt-bindings: misc: Add Movidius VPU coooling driver
 bindings

Add device tree bindings for Movidius VPU coooling driver

Signed-off-by: Demakkanavar, Kenchappa <kenchappa.demakkanavar@intel.com>
---
 .../bindings/misc/intel,vpu-cooling-dev.yaml  | 47 +++++++++++++++++++
 1 file changed, 47 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/misc/intel,vpu-cooling-dev.yaml

diff --git a/Documentation/devicetree/bindings/misc/intel,vpu-cooling-dev.yaml b/Documentation/devicetree/bindings/misc/intel,vpu-cooling-dev.yaml
new file mode 100644
index 000000000000..24b79cc80229
--- /dev/null
+++ b/Documentation/devicetree/bindings/misc/intel,vpu-cooling-dev.yaml
@@ -0,0 +1,47 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+# Copyright (C) 2020 Intel Corporation
+%YAML 1.2
+---
+$id: "http://devicetree.org/schemas/soc/intel/intel,vpu-cooling-dev.yaml#"
+$schema: "http://devicetree.org/meta-schemas/core.yaml#"
+
+title: Movidius VPU Cooling Device
+
+maintainers:
+  - Demakkanavar, Kenchappa <kenchappa.demakkanavar@intel.com>
+
+description:
+  Movidius VPU coooling driver is thermal cooling agent registered
+  to thermal framework. This driver communicates with vpu to get/set
+  power modes based on the thermal driver requested states.
+
+properties:
+  compatible:
+    const: intel,vpu-cooling-dev
+
+  intel,vpu-ipc-id:
+    $ref: "/schemas/types.yaml#/definitions/uint32"
+    description: The numeric ID identifying the VPU within the xLink stack.
+
+  #cooling-cells:
+    const: 2
+	description: If used as a cooling device, must be <2>
+    Also see:
+    Documentation/devicetree/bindings/thermal/thermal-cooling-devices.yaml
+    min and max states are derived from the speed-map of the fan..
+
+required:
+  - compatible
+  - intel,vpu-ipc-id:
+  - #cooling-cells
+
+additionalProperties: false
+
+examples:
+  - |
+    vpu-cooling-device {
+        compatible = "intel,vpu-cooling-dev";
+        intel,vpu-ipc-id = < 0x00 >;
+        status = "okay";
+        #cooling-cells = < 0x02 >;
+    };
-- 
2.25.1

